### PR TITLE
Added enableFeedback property to BottomNavigationBar

### PR DESCRIPTION
**Pull request by chinmoy12c**
## Description

Added enableFeedback property to BottomNavigationBar and BottomNavigationBarTheme.

## Related Issues

#69397 

## Tests

I added the following tests:

- BottomNavigationBar with enabled feedback.
- BottomNavigationBar with disabled feedback.
- BottomNavigationBar with enabled feedback by default.
- BottomNavigationBar with disabled feedback using BottomNavigationBarTheme.
- BottomNavigationBar.enableFeedback overrides BottomNavigationBarTheme.enableFeedback

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 1]
```diff
@@ -1828,6 +1829,139 @@ void main() {
...
+          home: Scaffold(
+            bottomNavigationBar: BottomNavigationBar(
+              enableFeedback: enableFeedback,
+              items: const <BottomNavigationBarItem>[
+                BottomNavigationBarItem(icon: Icon(Icons.ac_unit), title: Text('AC')),
+                BottomNavigationBarItem(icon: Icon(Icons.access_alarm), title: Text('Alarm')),
+              ],
+            ),
+          ),
+        ),
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart:1880]: This appears to be re-used multiple times, can be put in a method instead.
```suggestion
        MaterialApp(
          home: Scaffold(
            bottomNavigationBar: BottomNavigationBar(
              enableFeedback: enableFeedback,
              items: const <BottomNavigationBarItem>[
                BottomNavigationBarItem(icon: Icon(Icons.ac_unit), title: Text('AC')),
                BottomNavigationBarItem(icon: Icon(Icons.access_alarm), title: Text('Alarm')),
              ],
            ),
          ),
        ),
```

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 2]
```diff
@@ -1828,6 +1829,139 @@ void main() {
...
+      );
+
+      await tester.tap(find.byType(InkResponse).first);
+      await tester.pump(const Duration(seconds: 0));
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart:1884]: Why does this test pump by 0 seconds?

**Review by Piinks — COMMENTED** [review]
Hi @chinmoy12c thanks for the contribution!

File: packages/flutter/test/material/bottom_navigation_bar_test.dart (hunk 1 above)
Comment (chinmoy12c) [reply packages/flutter/test/material/bottom_navigation_bar_test.dart:1880]: Sure, I'll make the changes :)

File: packages/flutter/test/material/bottom_navigation_bar_test.dart (hunk 2 above)
Comment (chinmoy12c) [reply packages/flutter/test/material/bottom_navigation_bar_test.dart:1884]: Sorry, it's a mistake on my side, should have been 1. Also, using ``pumpAndSettle``` would be better. I'll change it.

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 3]
```diff
@@ -1828,6 +1829,107 @@ void main() {
...
+          bottomNavigationBar: BottomNavigationBar(
+            enableFeedback: enableFeedback,
+            items: const <BottomNavigationBarItem>[
+              BottomNavigationBarItem(icon: Icon(Icons.ac_unit), title: Text('AC')),
+              BottomNavigationBarItem(icon: Icon(Icons.access_alarm), title: Text('Alarm')),
+            ],
+          ),
+        ),
+      );
+    }
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart:1855]: Is this one necessary? The one with the theme below is just using null for `enableFeedbackTheme` unless it is set, which the default theme above would have. I don't think there needs to be two of these.

File: packages/flutter/test/material/bottom_navigation_bar_test.dart [hunk 4]
```diff
@@ -1828,6 +1829,107 @@ void main() {
...
+        enableFeedbackTheme: enableFeedbackTheme,
+        enableFeedback: enableFeedback
+      ));
+
+      await tester.tap(find.byType(InkResponse).first);
+      await tester.pumpAndSettle();
+      expect(feedback.clickSoundCount, 1);
+      expect(feedback.hapticCount, 0);
+    });
+  });
```
Comment (Piinks) [inline packages/flutter/test/material/bottom_navigation_bar_test.dart:1932]: Nit:
```suggestion
  });
  
```

**Review by Piinks — COMMENTED** [review]
Thanks for updating! It looks like this is just about ready! :)

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thanks for the contribution!

**Pull request merged**